Sample Code/SwiftSampleCode/DJISDKSwiftDemo/Missions/IOC/NavigationIOCViewController.swift

```diff
@@ -18,134 +19,101 @@ class NavigationIOCViewController: DJIBaseViewController, DJIFlightControllerDel
-        if aircraft != nil {
-        
-        
-        aircraft!.flightController?.setFlightOrientationMode(DJIFlightOrientationMode.DefaultAircraftHeading, withCompletion: {[weak self] (error: NSError?) -> Void in
-            if (error != nil) {
-                self?.showAlertResult("Stop IOC: \(error!.description)")
+        if let aircraft = self.fetchAircraft() {
+            aircraft.flightController?.setFlightOrientationMode(DJIFlightOrientationMode.DefaultAircraftHeading) { [weak self] (error: NSError?) -> Void in
+                if let error = error {
+                    self?.showAlertResult("Stop IOC: \(error.description)")
+                }
             }
-        })
         }
     }
 
     @IBAction func onRecordButtonClicked(sender: AnyObject) {
-        let aircraft: DJIAircraft? = self.fetchAircraft()
-        if aircraft == nil {
-            return
-        }
+        guard let aircraft = self.fetchAircraft() else { return }
         if self.isRecording {
-            aircraft!.camera!.stopRecordVideoWithCompletion({[weak self] (error: NSError?) -> Void in
-                if (error != nil ) {
-                    self?.showAlertResult("Stop Record: \(error!.description)")
-                } else {
+            aircraft.camera!.stopRecordVideoWithCompletion { [weak self] (error: NSError?) -> Void in
+                guard let error = error else {
                     self?.showAlertResult("Stop Record:Success")
+                    return
                 }
-            })
+                self?.showAlertResult("Stop Record: \(error.description)")
+            }
         }
         else {
-            aircraft!.camera!.startRecordVideoWithCompletion({[weak self] (error: NSError?) -> Void in
-                if (error != nil ) {
-                    self?.showAlertResult("Start Record: \(error!.description)")
-                } else {
+            aircraft.camera!.startRecordVideoWithCompletion { [weak self] (error: NSError?) -> Void in
+                guard let error = error else {
                     self?.showAlertResult("Start Record:Success")
-                }            })
+                    return
+                }
+                self?.showAlertResult("Start Record: \(error.description)")
+            }
         }
     }
 
     @IBAction func onLockCourseButtonClicked(sender: AnyObject) {
-        let aircraft: DJIAircraft? = self.fetchAircraft()
-        if aircraft == nil {
-            return
-        }
-        aircraft!.flightController?.lockCourseUsingCurrentDirectionWithCompletion({[weak self] (error: NSError?) -> Void in
-            
-            if (error != nil ) {
-                self?.showAlertResult("Lock Course: \(error!.description)")
-            } else {
+        guard let aircraft = self.fetchAircraft() else { return }
+        aircraft.flightController?.lockCourseUsingCurrentDirectionWithCompletion { [weak self] (error: NSError?) -> Void in
+            guard let error = error else {
```

Comment:
This makes sure there is an error. Guard should be used to protect the happy path.